proposals/lip-Introduce_Legacy_module.md

```diff
@@ -0,0 +1,277 @@
+
+
+##### Store Prefix, Store Key, and Store Value
+
+* The store prefix is set to `STORE_PREFIX_LEGACY_ACCOUNTS`.
+* The store key is an 8-byte value representing the legacy address.
+* The store value is set to the serialization using `legacyAccountsSchema` of the balance of the legacy account.
+
+
+##### JSON Schema
+
+```java
+legacyAccountsSchema = {
+    "type": "object",
+    "required": ["balance"],
+    "properties": {
+        "balance": {
+            "dataType": "uint64",
+            "fieldNumber": 1
+        }
+    }
+}
+```
+
+
+##### Properties and Default values
+
+This substore contains an entry for each legacy address for which no [reclaim transaction][LIP18-ReclaimTransaction] was included. 
+See also the [“Accounts without Public Key” section][LIP18-AccountsWithoutPKey] in LIP 0018. 
+
+The module store will be initialized with these legacy addresses and their respective balances during the execution of the genesis block.
+
+
+#### Internal Functions
+
+
+#### Legacy Addresses
+
+Obtaining the legacy 8-byte address from a public key.
+
+
+##### Parameters
+
+A 32-byte value representing a public key.
+
+
+##### Returns
+
+The 8-byte legacy address corresponding to the given input.
+
+
+##### Execution
+
+```python
+getLegacyAddress(publicKey):
+    hashedKey = SHA-256(publicKey)
+    firstEightBytes = 8 first bytes of hashedKey
+    reversedEightBytes = firstEightBytes reversed
+    return reversedEightBytes
+```
+
+
+### Commands
+
+
+#### Reclaim
+
+This command allows users to reclaim tokens from a legacy account as defined in [LIP 0018][LIP18-ReclaimTransaction]. Here, we clarify the verification and execution logic with respect to the module store.
+
+Transactions executing this command have:
+
+* `moduleID = MODULE_ID_LEGACY`,
+* `commandID = COMMAND_ID_RECLAIM`.
+
+
+##### Parameters
+
+The `params` property of a reclaim transaction must obey the following schema:
+
+```java
+reclaimParamsSchema = {
+    "type": "object",
+    "required": ["amount"],
+    "properties": {
+        "amount": {
+            "dataType": "uint64",
+            "fieldNumber": 1
+        }
+    }
+}
+```
+
+
+##### Verification
+
+For a reclaim transaction `trs` to be valid, the legacy accounts substore must contain an entry with store key equal to `getLegacyAddress(trs.senderPublicKey)` and store value `{"balance": trs.params.amount}`, serialized using `legacyAccountsSchema`.
+
+
+##### Execution
+
+When a reclaim transaction `trs` is executed, the following is done:
+
+* Delete the entry from the legacy accounts substore with store key `getLegacyAddress(trs.senderPublicKey)`.
+* Call the function `mint(newAddress, 0, trs.params.amount)`, where `newAddress` is the [20-byte address][LIP18-addressComputation] derived from `trs.senderPublicKey`. 
+  The function `mint` is defined in the [Token module][token-LIP].
```

Comment:
minor point, but maybe we could link directly to the function definition

Reply:
done